### ISSUE TITLE
MetricsLayer — Add additional event fields as KeyValue pairs to a recorded metric

### ISSUE THREAD

**Issue by jnicholls** [tokio-rs/tracing-opentelemetry#32]
## Feature Request

`MetricsLayer` — Add additional event fields as `KeyValue` pairs to a recorded metric.

### Motivation

`MetricsLayer` is a great idea, to have events be able to be captured and exported as metrics. Brilliant. But, the current implementation only submits the metric name and value, and no associated `KeyValue` pairs to add context to the metric.

### Proposal

It would be fantastic if other fields of the event were added as associated `KeyValue` pairs to the metric. Or, another specially named field specifically for `MetricsLayer` to inspect for optional `KeyValue` pairs.

**Comment by jtescher** [comment]
@jnicholls This sounds good! are you able to submit a PR to this effect?

**Comment by ymgyt** [comment]
If @jnicholls  doesn't mind, I'd like to try this feature.

My understanding is that in the [Metric api's Counter add operation spec](https://opentelemetry.io/docs/specs/otel/metrics/api/#add) says

> This API MUST accept the following parameter: 
* A numeric increment value.
* Attributes to associate with the increment value.

This feature allows you to associate the attributes with the metrics.

For example

```rust
tracing::info!
  monotonic_counter.foo = 1,
  key_1 = val_1,
  "foo",
);
```
would export metrics with the attributes of `key_1 = val_1`.

**Comment by jtescher** [comment]
Resolved in #43
